flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGate.java

```diff
@@ -68,7 +69,7 @@
  * will have an input gate attached to it. This will provide its input, which will consist of one
  * subpartition from each partition of the intermediate result.
  */
-public abstract class InputGate implements AutoCloseable {
+public abstract class InputGate implements AsyncDataInput<Optional<BufferOrEvent>>, AutoCloseable {
 
 	public static final CompletableFuture<?> AVAILABLE = CompletableFuture.completedFuture(null);
```

Comment:
The `InputGate#AVAILABLE`  should be removed because the new `AsyncDataInput` interface already has `AVAILABLE`.